### PR TITLE
Remove changelog and docs

### PR DESCRIPTION
**Pull request by brettlangdon**
We have decided to ditch the changelog.

This PR removes the changelog and removes references to it in the readme.

/cc @twolfson

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,14 +64,6 @@ If you are on OS X and running into issues (e.g. `globstar` isn't defined and `s
     - This installs `gsed` to your command line
 
 ### Releasing
-We have automated our release process to keep it consistent among developers. Before releasing, please make sure to update the `CHANGELOG.md`.
```
Comment (twolfson) [inline README.md:67]: The first sentence is still practical. Let's collapse the next "To publish" onto the first one to make a new paragraph:

``` md
We have automated our release process to keep it consistent among developers. To publish a new release, run:
```

Comment (brettlangdon) [inline README.md:67]: Whoops, went a little too aggressive. Thanks for catching.

**Comment by brettlangdon** [comment]
Put the first sentence back.

**Comment by twolfson** [comment]
:+1:

**Pull request merged**